### PR TITLE
YJIT: Move help descriptions to options.rs

### PR DESCRIPTION
**Pull request by k0kubun**
We sometimes forget adding entries to `yjit_options` or update defaults in it. To prevent that, this PR moves it to options.rs and puts it near `OPTIONS`.

### CODE REVIEW

File: ruby.c [hunk 1]
```diff
@@ -280,6 +280,23 @@ show_usage_line(const struct ruby_opt_message *m,
     }
 }
 
+#if USE_YJIT
+void
+rb_show_usage_line(const char *name, const char *description,
```
Comment (maximecb) [inline ruby.c:285]:
```suggestion
rb_print_usage_line(const char *name, const char *description,
```

**Comment by nobu** [comment]
What about https://github.com/nobu/ruby/pull/new/usage_line ?

File: yjit/src/options.rs [hunk 2]
```diff
@@ -231,3 +243,17 @@ pub fn parse_option(str_ptr: *const std::os::raw::c_char) -> Option<()> {
     // Option successfully parsed
     return Some(());
 }
+
+/// Print YJIT options for `ruby --help`.
+#[no_mangle]
+pub extern "C" fn rb_yjit_print_options(help: c_int, highlight: c_int, w: c_uint, columns: c_int) {
```
[the remark this answers is not in the document]
Comment (nobu) [reply yjit/src/options.rs:249]: It is the "width of option parts" or "indent width of descriptions".

File: ruby.c (hunk 1 above)
Comment (k0kubun) [reply ruby.c]: I want the functions names in ruby.c to be consistent with each other. It uses "show" in many places, so I changed the other name to be consistent with it instead https://github.com/ruby/ruby/pull/8583/commits/2132f3793f0a6ca63e286f4cdf7fc3e89caf9237.

File: yjit/src/options.rs (hunk 2 above)
[the remark this answers is not in the document]
Comment (k0kubun) [reply yjit/src/options.rs]: addressed in https://github.com/ruby/ruby/pull/8583/commits/7b8fb69d3d26c887ae2f462505981b506481a7ae.

**Review by maximecb — APPROVED** [review]

**Pull request merged**